### PR TITLE
Add CMS API

### PR DESCRIPTION
**Pull request by pallavigopi**
@tr0n-p46  please look at the descriptions for all of the parameters in `definitions/Content.yaml` . Also do cross check the request body for the POST request. I have incorporated everything I understood from the document. Please suggest necessary changes.

### CODE REVIEW

File: spec/definitions/Content.yaml [hunk 1]
```diff
@@ -0,0 +1,79 @@
+required:
+  - id
+type: object
+properties:
+  id:
+    title: Id
+    description: Unique content Identifier
+    type: string
```
Comment (tr0n-p46) [inline spec/definitions/Content.yaml:8]: `id` is an integer field.

File: spec/definitions/Content.yaml [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    title: Content download URL
+    description: URL to download the content
+    type: string
+    format: url
+  key:
+    title: Key
+    type: string
+    maxLength: 255
+    minLength: 1
+  is_dir:
```
Comment (tr0n-p46) [inline spec/definitions/Content.yaml:21]: Add default as false

File: spec/definitions/Content.yaml [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    title: Modified on
+    type: string
+    format: date-time
+    readOnly: true
+  enterprise:
+    title: Enterprise
+    description: URL of the enterprise resource
+    type: string
+    format: url
+  owner:
```
Comment (tr0n-p46) [inline spec/definitions/Content.yaml:74]: Owner is an object with
id: <integer>,
username: <string>

File: spec/definitions/Content.yaml [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    type: string
+    maxLength: 255
+    x-nullable: true
+  hash:
+    title: Content Hash
+    description: Hash string of the content
+    type: string
+  size:
+    title: Content Size
+    description: Size of the content
```
Comment (tr0n-p46) [inline spec/definitions/Content.yaml:36]: Add `in bytes` to description

File: spec/definitions/Content.yaml [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    title: Key
+    type: string
+    maxLength: 255
+    minLength: 1
+  is_dir:
+    title: Is Directory
+    type: boolean
+  kind:
+    title: Content kind
+    description: The data type of the content
```
Comment (tr0n-p46) [inline spec/definitions/Content.yaml:26]: In description write `mime type of the content`

File: spec/definitions/Content.yaml [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+    description: Unique content Identifier
+    type: string
+    format: uuid
+    readOnly: true
+  download_url:
+    title: Content download URL
+    description: URL to download the content
+    type: string
+    format: url
+  key:
```
Comment (tr0n-p46) [inline spec/definitions/Content.yaml:16]: `name` field is missing.

File: spec/paths/enterprise@{enterprise_id}@content@{content_id}@.yaml [hunk 7]
```diff
@@ -0,0 +1,73 @@
+get:
```
Comment (tr0n-p46) [inline spec/paths/enterprise@{enterprise_id}@content@{content_id}@.yaml:1]: All paths in content resource is v0 API's ie. of format
`https://{tenant}-api.esper.cloud/api/v0/enterprise/:id/content/`

File: spec/paths/enterprise@{enterprise_id}@content@.yaml [hunk 8]
```diff
@@ -0,0 +1,90 @@
+get:
+  operationId: getAllContent
+  summary: List content in the enterprise
+  description: Returns Content list
+  produces:
+    - application/json
+  parameters:
```
Comment (tr0n-p46) [inline spec/paths/enterprise@{enterprise_id}@content@.yaml:7]: We don't have support for filter's at the moment. However we do have search filter that may search in either name or tags or description.
for eg. `<content_api>/content/?search=test` returns all results that has string test in name, tags or description

File: spec/paths/enterprise@{enterprise_id}@content@.yaml [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+            x-nullable: true
+          results:
+            type: array
+            items:
+              $ref: '#/definitions/Content'
+  security:
+    - apiKey: []
+  tags:
+    - Content
+post:
```
Comment (tr0n-p46) [inline spec/paths/enterprise@{enterprise_id}@content@.yaml:57]: post is only allowed for 
`<content_api>/content/upload/` and content type is not application/json rather multipart/form-data
Please check application upload api `<application_api>/application/upload/` Both the same only key is different

File: spec/paths/enterprise@{enterprise_id}@content@.yaml [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+  operationId: postContent
+  summary: Add new content in the enterprise
```
Comment (tr0n-p46) [inline spec/paths/enterprise@{enterprise_id}@content@.yaml:59]: Change description to `Upload new content`

**Review by tarun91vas — APPROVED** [review]

**Review by tr0n-p46 — APPROVED** [review]

**Pull request merged**